### ISSUE TITLE
Both of the Tiff libraries that we use seem to be non-maintained

### ISSUE THREAD

**Issue by h-mayorquin** [catalystneuro/roiextractors#171]
While discussing this on nwbct:
https://github.com/catalystneuro/nwb-conversion-tools/pull/586

It turns out that both of the libraries that we use for extracting tiff seem to be non maintained:

https://pypi.org/project/scanimage-tiff-reader/#history
https://pypi.org/project/tiffile/#history

Given that PIL which is a well maintained library offers a tiff reader that is capable of reading both the data and the metadata I think we should build a tiff extractor based on it in case any of the two other libraries start failing or we notice a problem. We can then use any of those (`tiff-reader` and `tiffile`) libraries to provide memmap capabilities as this seems to be the one thing that we use them for in top of IO. 

Note that NEO is currently using that approach:
https://neo.readthedocs.io/en/stable/io.html#neo.io.TiffIO

**Comment by h-mayorquin** [comment]
Minan seems to have some utilities to read tiff lazily, writing here to check lately:
https://github.com/denisecailab/minian/blob/f2da3f9187dff9825f9ddeee6479306aeae5823e/minian/utilities.py

**Comment by pauladkisson** [comment]
Tiffile says it's deprecated and to use tifffile instead, but scanimage-tiff-reader just had a release in 2023, so I think that one is fine.

**Comment by bendichter** [comment]
Pillow does a decent job at reading tiffs

**Comment by bendichter** [comment]
Pillow does a decent job at reading tiffs

**Comment by h-mayorquin** [comment]
Yes, turns out tifffile is well maintained to the day and is still receiving updated.
